wallet/wallet.go

```diff
@@ -4195,6 +4196,91 @@ func (w *Wallet) ChainParams() *chaincfg.Params {
 	return w.chainParams
 }
 
+// scriptChangeSource is a ChangeSource which is used to
+// receive all correlated previous input value.
+type scriptChangeSource struct {
+	version uint16
+	script  []byte
+}
+
+func (src *scriptChangeSource) Script() ([]byte, uint16, error) {
+	return src.script, src.version, nil
+}
+
+func (src *scriptChangeSource) ScriptSize() int {
+	return len(src.script)
+}
+
+func makeScriptChangeSource(address string, version uint16, params *chaincfg.Params) (*scriptChangeSource, error) {
+	destinationAddress, err := dcrutil.DecodeAddress(address, params)
+	if err != nil {
+		return nil, err
+	}
+
+	var script []byte
+	if addr, ok := destinationAddress.(V0Scripter); ok && version == 0 {
```

Comment:
V0Scripter is a wallet-defined interface, implemented by some addresses it returns.  The address decoded by dcrutil will never implement this interface.